Source/app/routes.js

```diff
@@ -74,16 +74,26 @@ module.exports = function(app, passport) {
     app.get('/api/getBuildingData', function(req, res) {
         console.log(req.query);
         Building.findOne({_id : req.query._id})
-            .populate({
-                path: 'data_entries',
-                select: 'timestamp point'
-            }).lean()
-            .exec(function (err, building) {
+            .populate(
+                { path: 'data_entries',
+                    match : {timestamp : { $lt: "2018-03-15 00:45:00", $gte : "2018-03-14 21:00:00"}}, //THIS WORKS TO FILTER DATES
```

Comment:
Very nice! We could use this specific line to pass in two dates that the user wants to see data between.